spec/graphql/schema/input_object_spec.rb

```diff
@@ -733,4 +733,111 @@ class Mutation < GraphQL::Schema::Object
       assert_equal "NestedInputObjectSchema::ItemInput, NestedInputObjectSchema::ItemInput", res["data"]["nestedStuff"]["str"]
     end
   end
+
+  describe "generate_from_object_type" do
+    def get_field_type(field_type)
+      field_type.class == Class ? field_type : get_field_type(field_type.of_type)
+    end
+
+    it "is defined as a class method" do
+      assert_equal true, input_object.respond_to?(:generate_from_object_type) &&
+                 !input_object.instance_methods.include?(:generate_from_object_type)
+    end
+
+    it "expects only GraphQL::Schema::Object successor" do
+      assert_raises('Unexpected argument type, GraphQL::Schema::Object successor expected') {
+        input_object.generate_from_object_type(12) }
+
+    end
+
+    it "generates arguments based on provided object" do
+      class ExampleObjectType < GraphQL::Schema::Object
+        field :id, ID, null: false
+        field :points , Integer, null: true
+        field :updated_at, GraphQL::Types::ISO8601DateTime, null: false
+      end
+
+      class ExampleInput < GraphQL::Schema::InputObject
+        generate_from_object_type ExampleObjectType
+      end
```

Comment:
Instead of assigning and removing constants, how about assigning classes to local variables? Then we won't have to worry about naming conflicts later.

```suggestion
      example_object_type = Class.new(GraphQL::Schema::Object) do 
        field :id, ID, null: false
        field :points , Integer, null: true
        field :updated_at, GraphQL::Types::ISO8601DateTime, null: false
      end

      example_input = Class.new(GraphQL::Schema::InputObject)
        generate_from_object_type(example_object_type)
      end
```

Likewise for the tests below. What do you think?